tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -93,6 +93,7 @@ Value InsertDynamicAllocAndDealloc(Location loc, Value result,
 }
 
 Value InsertAllocAndDealloc(Location loc, Value result,
```

Comment:
Please rename this function, as it no longer inserts dealloc operations.

Reply:
Also, you need to fix the `Dynamic...` version of this function, too.

Reply:
Renaming is done. The current [BufferAssignment](https://github.com/tensorflow/tensorflow/pull/37212) doesn't support dynamically shaped types but in the near future, it will support it. If we use it now, the AllocOp positions would be wrong. Should we still update this function?

Reply:
No, we rely on this working. But supporting dynamic shapes should be high on the feature list.